utils/shared/src/main/scala/fastparse/ParserInput.scala

```diff
@@ -0,0 +1,92 @@
+package fastparse
+
+import scala.collection.mutable.ArrayBuffer
+
+abstract class ParserInputLogger {
+  def logDropBuffer(index: Int)
+  def logApply(index: Int)
+}
```

Comment:
I think we can remove this class, and leave it to who-ever wants logging (e.g. our own tests...) to inherit from `ParserInput` and override `dropBuffer` and `apply` themselves to do logging. It doesn't seem like a common enough thing that we need to provide a standalone class for people to work with